## 03_check_for_root.patch
## by Jonas Meurer <mejo@debian.org>
##
## check for UID == 0 before actually doing anything.

--- a/src/cryptsetup.c
+++ b/src/cryptsetup.c
@@ -6,6 +6,8 @@
 #include <errno.h>
 #include <unistd.h>
 #include <assert.h>
+#include <unistd.h>
+#include <sys/types.h>
 
 #include <libcryptsetup.h>
 #include <popt.h>
@@ -431,6 +433,10 @@
 
 int main(int argc, char **argv)
 {
+	if (geteuid() != 0) {
+		fprintf(stderr, _("You have to be root to use cryptsetup!\n"));
+		return EXIT_FAILURE;
+	}
 	static char *popt_tmp;
 	static struct poptOption popt_help_options[] = {
 		{ NULL,    '\0', POPT_ARG_CALLBACK, help, 0, NULL,                         NULL },
